New Data Source : oracle_db_system_versions (#31001)
terraform-providers/terraform-provider-azurerm: website/docs/d/oracle_database_system_versions.html.markdown

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: Data Source: azurerm_oracle_database_system_versions"
description: |-
  Gets information about existing Oracle Database Systems Versions.
---

# Data Source: azurerm_oracle_database_system_versions

Use this data source to access information about existing Oracle Database Systems Versions.

## Example Usage

```hcl

> data "azurerm_oracle_db_system_versions" "example" {

Reviewer: ```suggestion
data "azurerm_oracle_database_system_versions" "example" {
```

  location = "eastus"
}

output "id" {
  value = data.azurerm_oracle_db_system_versions.example.id
}
```

## Arguments Reference

The following arguments are supported:

* `location` - (Required) The Azure Region to query for the Oracle Database Systems Versions.

---

* `database_software_image_supported` - (Optional) Whether to filter the results to the set of Oracle Database versions that are supported for the database software images.

> * `database_system_shape` - (Optional) If provided, filters the results to the set of database versions which are supported for the given shape. Possible value is `VM.Standard.x86`.

Reviewer: ```suggestion
* `database_system_shape` - (Optional) If provided, filters the results to the set of database versions which are supported for the given shape. The only possible value is `VM.Standard.x86`.
```

* `shape_family` - (Optional) If provided, filters the results to the set of database versions which are supported for the given shape family. Possible values are  `EXADATA`, `EXADB_XS`, `SINGLENODE` and `VIRTUALMACHINE`.

* `storage_management` - (Optional) The DB system storage management option. Used to list database versions available for that storage manager. The only possible value is `LVM`.

> * `upgrade_supported` - (Optional) Whether to filter the results to the set of database versions which are supported for Upgrade.

Reviewer: ```suggestion
* `upgrade_supported` - (Optional) Whether to filter the results to the set of database versions which are supported for upgrade.
```

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported: 

> * `id` - The ID of the azurerm Oracle Database Systems Versions.

Reviewer: ```suggestion
* `id` - The ID of the Oracle Database Systems Versions.
```

---
subcategory: "Oracle"
layout: "azurerm"
page_title: "Azure Resource Manager: Data Source: azurerm_oracle_database_system_versions"
description: |-

>   Gets information about an existing  Oracle Database Systems Versions.

Reviewer: ```suggestion
  Gets information about existing Oracle Database Systems Versions.
```

---

# Data Source: azurerm_oracle_database_system_versions

Use this data source to access information about existing Oracle Database Systems Versions.

## Example Usage

```hcl
data "azurerm_oracle_db_system_versions" "example" {
  location = "eastus"
}

output "id" {
  value = data.azurerm_oracle_db_system_versions.example.id
}
```

## Arguments Reference

The following arguments are supported:

* `location` - (Required) The Azure Region to query for the Oracle Database Systems Versions.

---

> * `database_software_image_support_enabled` - (Optional) If true, filters the results to the set of Oracle Database versions that are supported for the database software images.

Reviewer: ```suggestion
* `database_software_image_filtered` - (Optional) Whether to filter the results to the set of Oracle Database versions that are supported for the database software images.
```

* `database_system_id` - (Optional) The DB system AzureId. If provided, filters the results to the set of database versions which are supported for the DB system.

* `database_system_shape` - (Optional) If provided, filters the results to the set of database versions which are supported for the given shape. e.g., VM.Standard.x86.

> * `shape_family` - (Optional) If provided, filters the results to the set of database versions which are supported for the given shape family.

Reviewer: The description should be `Possible values are ...` ([ref](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md#descriptions))

> * `storage_management` - (Optional) The DB system storage management option. Used to list database versions available for that storage manager. Valid values are ASM and LVM.

Reviewer: The description should be `Possible values are ...` ([ref](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-documentation-standards.md#descriptions))

* `upgrade_supported` - (Optional) If true, filters the results to the set of database versions which are supported for Upgrade.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported: 

* `id` - The ID of the azurerm Oracle Database Systems Versions.

* `versions` - A `versions` block as defined below.

---

A `versions` block exports the following:

* `latest_for_major_version` - Indicates if this version of the Oracle Database software is the latest version for a release.

> * `name` - Name of Oracle Database version.

Reviewer: ```suggestion
* `name` - The name of the Oracle Database version.

* `latest_for_major_version` - Indicates if this version of the Oracle Database software is the latest version for a release.
```

> * `name` - The name of Oracle Database version.

Reviewer: ```suggestion
* `name` - The name of the Oracle Database version.
```

* `supports_pdb_enabled` - Indicates if this version of the Oracle Database software supports pluggable databases.

> * `version` - A valid Oracle Database version.

Reviewer: same as [this comment](https://github.com/hashicorp/terraform-provider-azurerm/pull/31001#discussion_r2612562602)
